configuration/src/lib.rs

```diff
@@ -336,8 +336,11 @@ impl NomadConfig {
     }
 
     /// Get a reference to the nomad config's gas map.
-    pub fn gas(&self) -> &HashMap<String, NomadGasConfig> {
-        &self.gas
+    pub fn gas(&self) -> HashMap<String, &NomadGasConfig> {
+        self.gas
+            .iter()
+            .map(|(k, v)| (k.to_owned(), v.deref()))
```

Comment:
general rule `v.deref()` is usually better as `*v`

Reply:
Compiler isn't recognizing `*v` as `v.deref()` here. It's trying to deref `&NomadGasConfig[s]` into `NomadGasConfig[s]` not `NomadGasConfig`.

Reply:
This uses unary operator but it's ugly
```
self.gas
  .iter()
  .map(|(k, v)| -> (String, &NomadGasConfig) { (k.to_owned(), &*v) })
  .collect()
```